odoo/addons/base/models/res_partner.py

```diff
@@ -216,6 +216,8 @@ def default_get(self, default_fields):
     company_id = fields.Many2one('res.company', 'Company', index=True)
     color = fields.Integer(string='Color Index', default=0)
     user_ids = fields.One2many('res.users', 'partner_id', string='Users', auto_join=True)
+    main_user_id = fields.Many2one('res.users', string="Main User", compute='_compute_main_user', help="Useful to determine which user to use when a partner has multiple users.")
```

Comment:
I would avoid adding that kind of trick. Indeed people may use it and think a partner always has "a unique main user", which is wrong. People may have several internal users, maybe with their own calendar schedule and holidays (if a given user has several role in a company, and it exists as I already had to do bug fixes related to that case). I think it is better to have code correctly taking all users into consideration.

Reply:
What is the correct way to handle it though? Case by case?

For example for holidays, for a partner to be considered on leave, should it be "any user on leave" or "all users on leave"?
-> I would tend to think "all" in that case? Because the goal of leave is to tell the person is not available, so if any user is available then he's available.

For internal, is it "any user internal" or "all users internal"?
-> I guess "any" is enough? Because same reason, if any user is internal that should be enough to consider the partner has access to internal features.

For is public (technical) is it "any user is public" or "all user is public"?
-> As it's technical, I would say "any" is enough. (it is what is done currently)

Reply:
Done reverted that part to what was before, and used "all users must be on leave" for the purpose of leave.